submarine-server/server-core/src/main/java/org/apache/submarine/server/rest/ModelVersionRestApi.java

```diff
@@ -0,0 +1,295 @@
+    } catch (SubmarineRuntimeException e) {
+      return parseModelVersionServiceException(e);
+    }
+  }
+
+  /**
+   * Delete the model version with model version name and version.
+   *
+   * @param name    model version's name
+   * @param version model version's version
+   * @return seccess message
+   */
+  @DELETE
+  @Path("/{name}/{version}")
+  @Operation(summary = "Delete the model version", tags = {"model-version"}, responses = {
+      @ApiResponse(description = "successful operation",
+          content = @Content(schema = @Schema(implementation = JsonResponse.class))),
+      @ApiResponse(responseCode = "404", description = "ModelVersionEntity not found")})
+  public Response deleteModelVersion(@PathParam(RestConstants.MODEL_VERSION_NAME) String name,
+                                     @PathParam(RestConstants.MODEL_VERSION_VERSION) Integer version) {
+    try {
+      modelVersionService.delete(name, version);
+      return new JsonResponse.Builder<String>(Response.Status.OK).success(true)
+          .message("Delete the model version instance").build();
+    } catch (SubmarineRuntimeException e) {
+      return parseModelVersionServiceException(e);
+    }
+  }
+
+  /**
+   * Update the model version.
+   *
+   * @param entity model version entity
+   * @return success message
+   */
+  @PATCH
+  @Path("")
+  @Operation(summary = "Update the model version", tags = {"model-version"}, responses = {
+      @ApiResponse(description = "successful operation",
+          content = @Content(schema = @Schema(implementation = JsonResponse.class))),
+      @ApiResponse(responseCode = "404", description = "ModelVersionEntity not found")})
+  public Response updateModelVersion(ModelVersionEntity entity) {
+    try {
+      checkModelVersion(entity);
+      modelVersionService.update(entity);
+      return new JsonResponse.Builder<String>(Response.Status.OK).success(true)
+          .message("Update the model version instance").build();
+    } catch (SubmarineRuntimeException e) {
+      return parseModelVersionServiceException(e);
+    }
+  }
+
+  /**
+   * Create a model version tag.
+   *
+   * @param name    model version's name
+   * @param version model version's version
+   * @param tag     tag name
+   * @return success message
+   */
+  @POST
+  @Path("/tag")
+  @Consumes({ RestConstants.MEDIA_TYPE_YAML, MediaType.APPLICATION_JSON })
+  @Operation(summary = "Create a model version tag instance", tags = { "model-version" }, responses = {
+      @ApiResponse(description = "successful operation",
+          content = @Content(schema = @Schema(implementation = JsonResponse.class))) })
+  public Response createModelVersionTag(@DefaultValue("") @QueryParam("name") String name,
+                                        @DefaultValue("") @QueryParam("version") String version,
+                                        @DefaultValue("") @QueryParam("tag") String tag) {
```

Comment:
If a user does not pass enough parameters, server will return internal server error. Therefore, I add the default empty string, and check whether the values of parameters are not empty. This will return the actual error to the user.